deploy/eck-integrations/templates/eck-agent/elastic-agent-daemonset-secret.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: eck-agent-daemonset
+  namespace: {{ .namespace | default $.Release.Namespace | quote }}
+stringData:
+  agent.yml: |-
+    id: eck-agent-daemonset
+    outputs:
+      default:
+        type: elasticsearch
+        hosts:
+          - {{ .Values.elasticsearch.host | quote }}
+        username: {{ .Values.elasticsearch.user | quote  }}
+        password: {{ .Values.elasticsearch.pass | quote  }}
+    secret_references: []
+    agent:
+      download:
+        sourceURI: 'https://artifacts.elastic.co/downloads/'
+      monitoring:
+        namespace: {{ .Values.kubernetes.namespace }}
+        use_output: default
+        enabled: true
+        logs: true
+        metrics: true
+      providers:
+        kubernetes_leaderelection:
+          enabled: false
+        kubernetes:
+          node: ${NODE_NAME}
+          scope: node
+          hints.enabled: false
+          kubernetes_secrets:
+            enabled: true
+    inputs:
+    {{- include "agent.kubernetes.config.audit_logs_filestream.input" .Values.kubernetes | nindent 6 }}
+    {{- include "agent.kubernetes.config.container_logs.input" .Values.kubernetes | nindent 6 }}
+    {{- include "agent.kubernetes.config.kube_controller.input" .Values.kubernetes | nindent 6 }}
```

Comment:
Also there is a condition here:
            condition: '${kubernetes.labels.component} == ''kube-controller-manager'''